src/comm/Connection.h

```diff
@@ -77,10 +77,13 @@ class Connection: public CodeContext
     /** Clear the connection properties and close any open socket. */
     virtual ~Connection();
 
-    /** Copy an existing connections IP and properties.
-     * This excludes the FD. The new copy will be a closed connection.
-     */
-    ConnectionPointer copyDetails() const;
+    /// Create a new (closed) IDENT Connection object based on our from-Squid
+    /// connection properties.
+    ConnectionPointer cloneIdentDetails() const;
+
+    /// Create a new (closed) Connection object pointing to the same destination
+    /// as this from-Squid connection.
+    ConnectionPointer cloneDestinationDetails() const;
```

Comment:
FTR: From the unofficial commit 46f70173f1ab392bf3b8b6755f0fdfab8124e1e2 message, quoted here with minor edits/updates for clarity:

> I renamed copyDetails() because a reader like me would be puzzled by the copyDestinationDetails() vs. copyDetails() choice: Which one do I want? The new pair of names makes the choice easier. ... Finally, I prefer "clone" for methods that create new objects. The "copy" prefix is better suited for methods that fill existing objects.

Reply:
The reason you gave me for the insistence on clone vs copy years ago was that copy had a loose definition that implied some details were not copied, whereas clone implied a complete and absolute duplicate of the original object.

Using "clone" methods to only duplicate parts of objects goes against that principle.

Reply:
I do not mind renaming the 'clone' prefix to 'copy' in both these methods.

Reply:
> The reason you gave me for the insistence on clone vs copy years ago was that copy had a loose definition that implied some details were not copied, whereas clone implied a complete and absolute duplicate of the original object.

Bare clone() and bare copy() would indeed imply "full" replication, but we are not dealing with bare clone() and bare copy() method names here, only prefixes. The "clone" prefix is a better fit here because (as documented) we are creating a new object as opposed to filling in an existing one.

If "years ago" I said and meant something that contradicts the above, then I was wrong.

> I do not mind renaming the 'clone' prefix to 'copy' in both these methods.

I recommend using "clone", but if Amos insists on "copy" for some reason, then I am not going to fight about it.

Reply:
I'm not particularly attached to either naming myself. Just wanted to point out the discontinuity with the arguments over why the original method was not allowed to be called cloneDetails().